lib/ansible/playbook/play_context.py

```diff
@@ -434,8 +434,12 @@ def set_task_and_variable_override(self, task, variables, templar):
         # set become defaults if not previouslly set
         task.set_become_defaults(new_info.become, new_info.become_method, new_info.become_user)
 
+        if task.check_mode is not None:
```

Comment:
you still don't deal with `check_mode: False` and now always_run will override check_mode setting, I would reverse it in view of deprecating always_run